client/my-sites/preview/main.jsx

```diff
@@ -124,6 +130,27 @@ class PreviewMain extends React.Component {
 		return this.props.site.options.unmapped_url;
 	}
 
+	showEditButton = () => {
+		if ( 'posts' === this.props.site.options.show_on_front ) {
```

Comment:
`site` may be null on a cold cache. Would maybe recommend a lodash get here `get( site, [ 'options', 'show_on_front' ] )` if there isn't a closer global selector available.